src/Zio/UPathExtensions.cs

```diff
@@ -239,28 +239,26 @@ public static bool IsInDirectory(this UPath path, UPath directory, bool recursiv
     /// </summary>
     /// <param name="path">The path.</param>
     /// <param name="name">The name of a parameter to include n the <see cref="ArgumentNullException"/>.</param>
-    /// <returns>A path not modified.</returns>
     /// <exception cref="System.ArgumentNullException">If the path was null using the parameter name from <paramref name="name"/></exception>
-    public static UPath AssertNotNull(this UPath path, string name = "path")
```

Comment:
I would prefer that we do not change these public API (`AssertNotNull` and `AssertAbsolute`) so we keep the behavior. They can be marked with inlining, but I doubt that these would show up as bottleneck in a profiler.

Reply:
AssertAbsolute returned upath.Fullname, hence went through UPath ctor and the normalization, which is a rather expensive operation to do if only 1 path in the code base used it, and could've easily be switched over.

If you want to keep the public API as is, there could be 2 alternatives:
- Add some internal ones without returning the path, and call them from the public one
- Get somehow rid of the string -> upath conversion in the AssertAbsolute method.

Reply:
> AssertAbsolute returned upath.Fullname, hence went through UPath ctor and the normalization, which is a rather expensive operation to do if only 1 path in the code base used it, and could've easily be switched over.

Oh, I don't remember why it was doing that, but I would believe it was an oversight and it should only return `path` instead.

Reply:
reverted the change and adjusted to return path.